### PR TITLE
[TEVA-926] Autocomplete tests

### PR DESCRIPTION
**Pull request by alexbowen**
adds unit tests for autocomplete view

https://dfedigital.atlassian.net/browse/TEVA-926

### CODE REVIEW

File: app/frontend/src/lib/__mocks__/autocomplete.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+    show: jest.fn(),
+    hide: jest.fn(),
+    focus: jest.fn()
+}
+
+const autocomplete = {
+    view
+}
+
+export default autocomplete
```
Comment (cpjmcquillan) [inline app/frontend/src/lib/__mocks__/autocomplete.js:11]: Newline. And this file looks 4 space indented

Comment (alexbowen) [reply app/frontend/src/lib/__mocks__/autocomplete.js:11]: ah this file isnt actually needed

File: app/frontend/src/lib/autocomplete.test.js [hunk 2]
```diff
@@ -1,34 +1,150 @@
...
+import autocomplete, { isActive, getOptions, renderAutocomplete } from './autocomplete';
+import view, { show, hide } from './autocomplete.view';
 
-describe('isActive', () => {
-  test('should activate autocomplete if threshold has been met', () => {
-    expect(isActive(3, 'sou')).toBe(true);
-    expect(isActive(3, 'sout')).toBe(true);
+describe('autocomplete', () => {
+  describe('isActive', () => {
+    test('should activate autocomplete if threshold has been met', () => {
```
Comment (cpjmcquillan) [inline app/frontend/src/lib/autocomplete.test.js:6]: activates instead of 'should..'

File: app/frontend/src/lib/autocomplete.test.js [hunk 3]
```diff
@@ -1,34 +1,150 @@
...
+      expect(isActive(3, 'sou')).toBe(true);
+      expect(isActive(3, 'sout')).toBe(true);
+    });
+
+    test('should not activate autocomplete if threshold hasnt been met', () => {
```
Comment (cpjmcquillan) [inline app/frontend/src/lib/autocomplete.test.js:11]: does not activate

File: app/frontend/src/lib/autocomplete.test.js [hunk 4]
```diff
@@ -1,34 +1,150 @@
...
+    'apple',
+    'banana',
+    'apple apple',
+    'banana apple',
+    'applebanana',
+    'cherry',
+  ];
+
+  describe('getOptions', () => {
+    test('should return an array of matches from the options array that contain the supplied string', () => {
```
Comment (cpjmcquillan) [inline app/frontend/src/lib/autocomplete.test.js:27]: returns

File: app/frontend/src/lib/autocomplete.test.js [hunk 5]
```diff
@@ -1,34 +1,150 @@
...
+      expect(getOptions(options, 'appl')).toEqual(['apple', 'apple apple', 'banana apple', 'applebanana']);
+      expect(getOptions(options, 'a')).toEqual(['apple', 'banana', 'apple apple', 'banana apple', 'applebanana']);
+    });
+
+    test('should return an array of matches from the options irrespective of letter case', () => {
```
Comment (cpjmcquillan) [inline app/frontend/src/lib/autocomplete.test.js:32]: does not return

**Pull request merged**